cpp/dolfinx/fem/CoordinateElement.cpp

```diff
@@ -49,12 +47,20 @@ void CoordinateElement::push_forward(
     const Eigen::Ref<const Eigen::Array<double, Eigen::Dynamic, Eigen::Dynamic,
                                         Eigen::RowMajor>>& cell_geometry) const
 {
-  assert(_compute_physical_coordinates);
+  assert(_finite_element);
   assert(x.rows() == X.rows());
   assert(x.cols() == _gdim);
   assert(X.cols() == _tdim);
-  _compute_physical_coordinates(x.data(), X.rows(), X.data(),
-                                cell_geometry.data());
+
+  // Compute physical coordinates
+  Eigen::Tensor<double, 3, Eigen::RowMajor> phi(1, X.rows(),
+                                                cell_geometry.rows());
+  Eigen::Map<
```

Comment:
Is there a neater  way to do this with Eigen?

Reply:
Maybe. There is `.slice` and `.chip` - I will try out.